app/Helpers/shortcode.php

```diff
@@ -75,11 +75,11 @@ function normalize_user_shortcodes(string $value): string
      * find any url variants of user links and transform them into tags
      */
     $find = [
-        "~\<a [^/>]*retroachievements\.org/user/([\w]{1,20})[^/>]*\][^</a>]*</a>~si",
-        "~\[url[^\]]*retroachievements\.org/user/([\w]{1,20})[^\]]*\][^\[]*\[/url\]~si",
-        "~\[url[^\]]*?user/([\w]*).*?\[/url\]~si",
-        "~https?://(?:[\w\-]+\.)?retroachievements\.org/user/([\w]{1,20})~si",
-        "~https?://localhost(?::\d{1,5})?/user/([\w]{1,20})~si",
+        "~\<a [^/>]*retroachievements\.org/user/([\w]{1,20})(?![\w/])[^/>]*\][^</a>]*</a>~si",
+        "~\[url[^\]]*retroachievements\.org/user/([\w]{1,20})(?![\w/])[^\]]*\][^\[]*\[/url\]~si",
+        "~\[url[^\]]*?user/([\w]{1,20})(?![\w/]).*?\[/url\]~si",
+        "~https?://(?:[\w\-]+\.)?retroachievements\.org/user/([\w]{1,20})(?![\w/])~si",
+        "~https?://localhost(?::\d{1,5})?/user/([\w]{1,20})(?![\w/])~si",
```

Comment:
Not sure if you want to bother, but It's doing the same thing with parameters:
```
http://localhost/user/Jamiras?g=15
```
![image](https://github.com/RetroAchievements/RAWeb/assets/32680403/91d1b31f-37e7-4d04-8e4c-cf2a1bbcc095)

Reply:
Good catch. Should be taken care of now. Here is a sample post:

```
https://retroachievements.org/user/Jamiras/

https://retroachievements.org/user/Jamiras

https://retroachievements.org/user/Jamiras?g=15

https://retroachievements.org/user/Jamiras/progress
```